tutorials/tmva/TMVA_CNN_Classification.C

```diff
@@ -461,6 +461,21 @@ void TMVA_CNN_Classification(std::vector<bool> opt = {1, 1, 1, 1})
       }
    }
 
+   if (usePyTorchCNN) {
+
+      Info("TMVA_CNN_Classification", "Using Convolutional PyTorch Model");
+      // execute
+      gSystem->Exec("python PyTorch_Generate_CNN_Model.py");
```

Comment:
I would do a check if the file exists and modify the code as following:
```
   if (usePyTorchCNN) {

      Info("TMVA_CNN_Classification", "Using Convolutional PyTorch Model");
      // execute Python code for Pytorch
      TString pytorchFileName = "PyTorch_Generate_CNN_Model.py";
      if (gSystem->AccessPathName(pytorchFileName)) {
         Error("TMVA_CNN_Classification","Python file %s defining the PyTorch model is missing - skip using PyTorch",pytorchFileName.Data());
      }
      else {
         gSystem->Exec(TString::Format("python %s",pytorchFileName.Data()) );

         // book PyKeras method only if Keras model could be created
         Info("TMVA_CNN_Classification", "Booking PyTorch CNN model");
         factory.BookMethod(
            loader, TMVA::Types::kPyTorch, "PyTorch",
            "H:!V:VarTransform=None:FilenameModel=PyTorchModelCNN.pt:"
            "FilenameTrainedModel=PyTorchTrainedModelCNN.pt:NumEpochs=20:BatchSize=100"
            ":UserCode=PyTorch_Generate_CNN_Model.py");
      }
   }
```

Is it necessary to execute the python code `"PyTorch_Generate_CNN_Model.py` before in the C++ macro when we will also execute later in the MethodPyTorch class when we set up a model ? 

However after trying several executing the script from the MethodPyTorch class fails. Here is the error message:

```
Info in <TMVA_CNN_Classification>: Using Convolutional PyTorch Model
Info in <TMVA_CNN_Classification>: Booking PyTorch CNN model
Factory                  : Booking method: PyTorch
                         : 
                         : Using PyTorch - setting special configuration options 
                         : Using PyTorch version 1
                         :  Setup PyTorch Model 
                         :  Executing user initialization code from  PyTorch_Generate_CNN_Model.py
<WARNING>                : Failed to run python code: exec(open('PyTorch_Generate_CNN_Model.py').read())
<WARNING>                : Python error message:
Traceback (most recent call last):
  File "/home/moneta/.local/lib/python3.6/site-packages/torch/_utils_internal.py", line 49, in get_source_lines_and_file
    sourcelines, file_lineno = inspect.getsourcelines(obj)
  File "/usr/lib/python3.6/inspect.py", line 955, in getsourcelines
    lines, lnum = findsource(object)
  File "/usr/lib/python3.6/inspect.py", line 786, in findsource
    raise OSError('could not get source code')
OSError: could not get source code

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 113, in <module>
  File "/home/moneta/.local/lib/python3.6/site-packages/torch/jit/_script.py", line 898, in script
    obj, torch.jit._recursive.infer_methods_to_compile
  File "/home/moneta/.local/lib/python3.6/site-packages/torch/jit/_recursive.py", line 352, in create_script_module
    return create_script_module_impl(nn_module, concrete_type, stubs_fn)
  File "/home/moneta/.local/lib/python3.6/site-packages/torch/jit/_recursive.py", line 406, in create_script_module_impl
    script_module = torch.jit.RecursiveScriptModule._construct(cpp_module, init_fn)
  File "/home/moneta/.local/lib/python3.6/site-packages/torch/jit/_script.py", line 388, in _construct
    init_fn(script_module)
  File "/home/moneta/.local/lib/python3.6/site-packages/torch/jit/_recursive.py", line 388, in init_fn
    scripted = create_script_module_impl(orig_value, sub_concrete_type, stubs_fn)
  File "/home/moneta/.local/lib/python3.6/site-packages/torch/jit/_recursive.py", line 364, in create_script_module_impl
    method_stubs = stubs_fn(nn_module)
  File "/home/moneta/.local/lib/python3.6/site-packages/torch/jit/_recursive.py", line 591, in infer_methods_to_compile
    stubs.append(make_stub_from_method(nn_module, method))
  File "/home/moneta/.local/lib/python3.6/site-packages/torch/jit/_recursive.py", line 52, in make_stub_from_method
    return make_stub(func, method_name)
  File "/home/moneta/.local/lib/python3.6/site-packages/torch/jit/_recursive.py", line 37, in make_stub
    ast = get_jit_def(func, name, self_name="RecursiveScriptModule")
  File "/home/moneta/.local/lib/python3.6/site-packages/torch/jit/frontend.py", line 197, in get_jit_def
    sourcelines, file_lineno, filename = get_source_lines_and_file(fn, torch._C.ErrorReport.call_stack())
  File "/home/moneta/.local/lib/python3.6/site-packages/torch/_utils_internal.py", line 56, in get_source_lines_and_file
    raise OSError(msg) from e
OSError: Can't get source for <bound method Reshape.forward of Reshape()>. TorchScript requires source access in order to carry out compilation, make sure original .py files are available.
<FATAL>                         : Error executing the provided user code
```
@AnirudhDagar  do you know how to fix this ?

Reply:
@omazapa thanks for the fix. This should be resolved with your PR.